src/renderer/msg_filter.rs

```diff
@@ -0,0 +1,74 @@
+//! This module contains logic for filtering windows messages.
+
+use bitflags::bitflags;
+use windows::Win32::UI::WindowsAndMessaging::*;
+
+bitflags! {
+    /// Flags for filtering window messages.
+    #[repr(transparent)]
+    pub struct MessageFilter: u32 {
+        /// Blocks keyboard input event messages.
+        const InputKeyboard = 1u32 << 0;
+        /// Blocks mouse input event message.
+        const InputMouse = 1u32 << 1;
+        /// Blocks raw input event messages.
+        const InputRaw = 1u32 << 2;
+
+        /// Blocks window gain/lose focus event messages.
+        const WindowFocus = 1u32 << 8;
+        /// Blocks window control event messages
+        /// like move, resize, minimize, etc.
+        const WindowControl = 1u32 << 9;
+        /// Blocks window close messages.
+        const WindowClose = 1u32 << 10;
+
+        /// Blocks messages ID from 0 to WM_USER - 1
+        /// (the range for system-defined messages).
+        const RangeSystemDefined = 1u32 << 28;
+        /// Blocks messages ID from WM_USER to WM_APP - 1
+        /// (the range for private window classes like form button).
+        const RangePrivateReserved = 1u32 << 29;
+        /// Blocks messages ID from WM_APP to 0xBFFF
+        /// (the range for internal use of user application).
+        const RangeAppPrivate = 1u32 << 30;
+        /// Blocks messages ID from 0xC000 tp 0xFFFF
+        /// (the range for registered use between user applications).
+        const RangeAppRegistered = 1u32 << 31;
+
+        /// Blocks keyboard, mouse, raw input messages.
+        const InputAll = Self::InputKeyboard.bits() | Self::InputMouse.bits() | Self::InputRaw.bits();
+        /// Blocks window focus, control, close messages.
+        const WindowAll = Self::WindowFocus.bits() | Self::WindowControl.bits() | Self::WindowClose.bits();
+    }
+}
+
+impl MessageFilter {
+    /// Check whether the message ID is blocked by this filter
+    pub fn is_blocking(&self, message_id: u32, _wparam: usize, _lparam: isize) -> bool {
```

Comment:
`_wparam` and `_lparam` aren't used here. Considering that this is a public facing API, this will be confusing to users. Could you please remove them?

Reply:
It was "Here's your starting point if you want to fork and implement custom complex filter" kind of parameters,
but I'm not confident about its necessity either.
If you don't agree with its reason of existance, I'll remove it. Please let me know.

Reply:
I don't think ease of forking should be a goal. It would be about a separate project after all, in a sense, and it's unreasonable to optimize for that and document that. Plus, it might be confusing to the readers of the code. Yeah, I'd say let's remove them. Thank you!